src/test/java/org/opensearch/neuralsearch/plugin/query/NeuralQueryBuilderTests.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.plugin.query;
+
+import static org.opensearch.index.query.AbstractQueryBuilder.BOOST_FIELD;
+import static org.opensearch.index.query.AbstractQueryBuilder.NAME_FIELD;
+import static org.opensearch.neuralsearch.plugin.TestUtils.xContentBuilderToMap;
+import static org.opensearch.neuralsearch.plugin.query.NeuralQueryBuilder.K_FIELD;
+import static org.opensearch.neuralsearch.plugin.query.NeuralQueryBuilder.MODEL_ID_FIELD;
+import static org.opensearch.neuralsearch.plugin.query.NeuralQueryBuilder.NAME;
+import static org.opensearch.neuralsearch.plugin.query.NeuralQueryBuilder.QUERY_TEXT_FIELD;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.common.ParsingException;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class NeuralQueryBuilderTests extends OpenSearchTestCase {
+
+    private static final String FIELD_NAME = "testField";
+    private static final String QUERY_TEXT = "Hello world!";
+    private static final String MODEL_ID = "mfgfgdsfgfdgsde";
+    private static final int K = 10;
+    private static final float BOOST = 1.8f;
+    private static final String QUERY_NAME = "queryName";
+
+    public void testFromXContent_valid_withDefaults() throws IOException {
+        /*
+          {
+              "VECTOR_FIELD": {
+                "query_text": "string",
+                "model_id": "string",
+                "k": int
+              }
+          }
+        */
+        XContentBuilder xContentBuilder = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject(FIELD_NAME)
+            .field(QUERY_TEXT_FIELD.getPreferredName(), QUERY_TEXT)
+            .field(MODEL_ID_FIELD.getPreferredName(), MODEL_ID)
+            .field(K_FIELD.getPreferredName(), K)
+            .endObject()
+            .endObject();
+
+        XContentParser contentParser = createParser(xContentBuilder);
+        NeuralQueryBuilder neuralQueryBuilder = NeuralQueryBuilder.fromXContent(contentParser);
+
+        assertEquals(FIELD_NAME, neuralQueryBuilder.getFieldName());
+        assertEquals(QUERY_TEXT, neuralQueryBuilder.getQueryText());
+        assertEquals(MODEL_ID, neuralQueryBuilder.getModelId());
+        assertEquals(K, neuralQueryBuilder.getK());
+    }
+
+    public void testFromXContent_valid_withOptionals() throws IOException {
```

Comment:
see if we want to rename the test function to something like testFromXConent_whenXXXX_thenYYYYY for all the functions. This can help in just getting to know what is being tested with what just be reading the name of the fuction.

Reply:
Sure, will update